addon/services/ajax.js

```diff
@@ -113,9 +113,9 @@ export default Ember.Service.extend({
         );
 
         if (response instanceof AjaxError) {
-          reject(response);
+          Ember.run.join(null, reject, response);
```

Comment:
> @rwjblue create wait() for this purpose. This helper relies on the run loop to know if all scheduled timers have been cleared before resolving the wait()

this isn't correct,  `wait` doesn't detect run-lops rather explicitly detects promises async https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/ext/rsvp.js#L28